CRM/Contribute/BAO/Contribution.php

```diff
@@ -2755,6 +2755,10 @@ public static function recordFinancialAccounts(&$params, $financialTrxnValues =
         $balanceTrxnParams['status_id'] = $statusId;
         $balanceTrxnParams['payment_instrument_id'] = $params['contribution']->payment_instrument_id;
         $balanceTrxnParams['check_number'] = CRM_Utils_Array::value('check_number', $params);
+        if (!empty($balanceTrxnParams['from_financial_account_id']) && 
+            ($statusId == array_search('Completed', $contributionStatuses) || $statusId == array_search('Partially paid', $contributionStatuses))) {
+          $balanceTrxnParams['is_payment'] = 1;
+        }
```

Comment:
So if i am following correctly is_payment will be set to TRUE only if Contribution is partially paid. Not in case of full payment